### ISSUE TITLE
implementation of Mean Teacher

### ISSUE THREAD

**Issue by hzhz2020** [google-research/fixmatch#74]
In your implementation of Mean Teacher, are the student model and teacher model have exactly the same weights? While in original Mean Teacher paper, teacher model is EMA of student model weights.

**Comment by carlini** [comment]
Why do you think this is the case?

Looking here
https://github.com/google-research/fixmatch/blob/d4985a158065947dba803e626ee9a6721709c570/mean_teacher.py#L58

the teacher's logits are accessed through the `ema_getter`, which means it's accessing the EMA weights and not the current weights.

**Comment by hzhz2020** [comment]
i overlooked that. thanks!
